### PR TITLE
Fix: RBAC Same role as creator list view data display as - (dash)

### LINKED ISSUE #10010: Bug with RBAC - custom roles. Users with "has same role as creator" don't see content in ListView.
### Describe the bug

When using a custom role with the defined condition of "**Has same role as creator**", the users with said role are unable to view the content in **ListView** which was created by another user with the same role. The entries created by the other users do show up, but all the custom fields other than the auto generated ones (created_at, published_at, etc) are populated with **'-'** instead of the actual data. Clicking on the entries to modify, however does show all the fields.

### Steps to reproduce the behavior

1. Create a brand new project with `yarn create strapi-app my-project --quickstart`
2. Add the EE license in to `.env` or `license.txt` (I used a trial license)
3. Rebuild the admin panel with `yarn build --clean`
4. Start the project with `yarn develop`
5. Create a sample model (called `article` perhaps) with a couple of fields (for eg: `title`, `content`, `isActive`)
**title**: string
**body**: richtext
**isActive**: Boolean
6. Create a new role `test`
7. Assign all field permissions but apply custom conditions
        **Can Create**: Anytime
        **Can Read**: Is creator and Has same role as creator
        **Can Update**: Is creator
        **Can Delete**: Is creator
8. Create 2 test users (User One and User two)
9. Create an article on one
10. Attempt to read on the other
### Expected behavior

Should be able to view all the content in the ListView

### Screenshots
- User One creates the article; Can view it
image
- User Two attempts to read the article; Cannot view content. (Although the entry for it remains visible)
image

### System

- Node.js version: v14.15.4
- NPM version: 6.14.10
- Strapi version: 3.5.4
- Database: SQLite (but should be reproducable with others)
- Operating system: Windows 10 with WSL

### Additional context
[#8129](https://github.com/strapi/strapi/issues/8129)
Similar instance of the problem

### PR DESCRIPTION
**Pull request by PatrickJira**
Fix #10010 v4

### What does it do?

Fix RBAC bug in list view that show dash data for entries created by `same role as creator`. From debug, problem seems to came from casl v5 `permittedFieldsOf` not working as how it should be.
The fix change to pass subject as string instead of entry object that marked type by casl

### Why is it needed?

To fix list view not working as intended when config RBAC with `can read` + `same role as creator`

### How to test it?

1. Create fresh project quickstart / sqlite
2. Create a collection type model with a field, for example **title: string**
3. Create a new role `test`
4. Assign all field permissions and apply conditions
  - **Can Create**: blank
  - **Can Read**: `Is creator` and `Has same role as creator`
  - **Can Update**: `Is creator`
  - **Can Delete**: `Is creator`
5. Create 2 users with `test` role
6. Login as one, create an entry
7. Login as another one, see content in list view. You should see content normally

### Related issue(s)/PR(s)

Issue #10010 but for v4

### CODE REVIEW

**Review by Convly — APPROVED** [review]
Seems to work well.
LGTM, waiting for the tests to pass

**Comment by innerdvations** [comment]
Looks like you just need to add the same fix to the new validate as well, validate.js line 133: https://github.com/strapi/strapi/blob/a2b1a1e72400fea121da3c3fb18f524b0c885ba8/packages/core/admin/server/services/permission/permissions-manager/validate.js#L133

**Comment by PatrickJira** [comment]
Update as @innerdvations suggestion

**Comment by PatrickJira** [comment]
My PR seems to have problem with permission ability that has fields value(s) as condition. So I need to do another approach to fix this.

**Comment by PatrickJira** [comment]
Problem lies within the list view data that doesn't fetch deep into role relation, which IMO may cause performance issue.
The permission ability,
```javascript
{
  "action": "plugin::content-manager.explorer.read",
  "subject": "api::article.article",
  "fields": [
    "title"
  ],
  "conditions": {
    "$and": [
      {
        "$or": [
          {
            "createdBy.id": 7
          },
          {
            "createdBy.roles": {
              "$elemMatch": {
                "id": {
                  "$in": [
                    7
                  ]
                }
              }
            }
          }
        ]
      },
      {
        "locale": {
          "$in": [
            "en",
            "th"
          ]
        }
      }
    ]
  }
}

```

**Comment by PatrickJira** [comment]
This is proper fix, BUT may cause performance issue.

Or I should create another helper specific for `find` that will skip conditions check and only do fields check. It'll not sacrifice the performance. See [this](https://github.com/PatrickJira/strapi/commit/4de5da7d1c63e165ab7035bdaad5337196361a4b)

File: packages/core/content-manager/server/controllers/collection-types.js [hunk 1]
```diff
@@ -23,6 +23,7 @@ module.exports = {
     const permissionQuery = await permissionChecker.sanitizedQuery.read(query);
 
     const populate = await getService('populate-builder')(model)
+      .populateFromQuery(permissionQuery)
```
Comment (PatrickJira) [inline packages/core/content-manager/server/controllers/collection-types.js:26]: If there are permission with field condition(s) check AFTER query. Permission service will need these fields to do the job, or else it deny all output fields

**Comment by riverus-dev** [comment]
Hi there! Which is the current status of this?

**Comment by innerdvations** [comment]
> This is proper fix, BUT may cause performance issue.

This should be fine for performance, it's how all the other methods are already done. However, I do think there are some improvements that could be made universally in how this is populated that we can address later. Thanks!

**Review by innerdvations — APPROVED** [review]

**Pull request merged**